src/native/libs/System.Globalization.Native/pal_calendarData.m

```diff
@@ -141,4 +170,128 @@
+                    eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+                    currentEra = [eraComponents era];
+                    if (currentEra != era)
+                    {
+                        // add back 1 day to get back into the specified Era
+                        startDateComponents.day = startDateComponents.day + 1;
+                        date = [japaneseCalendar dateFromComponents:startDateComponents];
+                        NSCalendar *gregorianCalendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
+                        NSDateComponents *components = [gregorianCalendar components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:date];
+                        *startYear = [components year];
+                        *startMonth = [components month];
+                        *startDay = [components day];
+                        return 1;
+                    }
+                }
+            }
+            // add 1 month at a time until we get into the specified Era
+            startDateComponents.month = startDateComponents.month + 1;
+            date = [japaneseCalendar dateFromComponents:startDateComponents];
+            eraComponents = [japaneseCalendar components:NSCalendarUnitEra fromDate:date];
+            currentEra = [eraComponents era];
+        }
+
+        return 0;
+    }
+}
+
+/*
+Function:
+GetCalendarsNative
+
+Returns the list of CalendarIds that are available for the specified locale.
+*/
+int32_t GlobalizationNative_GetCalendarsNative(const char* localeName, CalendarId* calendars, int32_t calendarsCapacity)
+{
+    @autoreleasepool
+    {
+        NSArray *calendarIdentifiers = @[
+            NSCalendarIdentifierGregorian,
+            NSCalendarIdentifierBuddhist,
+            NSCalendarIdentifierChinese,
+            NSCalendarIdentifierCoptic,
+            NSCalendarIdentifierEthiopicAmeteMihret,
+            NSCalendarIdentifierEthiopicAmeteAlem,
+            NSCalendarIdentifierHebrew,
+            NSCalendarIdentifierISO8601,
+            NSCalendarIdentifierIndian,
+            NSCalendarIdentifierIslamicCivil,
+            NSCalendarIdentifierIslamicTabular,
+            NSCalendarIdentifierIslamicUmmAlQura,
+            NSCalendarIdentifierIslamic,
+            NSCalendarIdentifierJapanese,
+            NSCalendarIdentifierPersian,
+            NSCalendarIdentifierRepublicOfChina,
+        ];
+
+        NSString *locName = [NSString stringWithFormat:@"%s", localeName];
+        NSLocale *currentLocale = [[NSLocale alloc] initWithLocaleIdentifier:locName];
+        NSString *defaultCalendarIdentifier = [currentLocale calendarIdentifier];
+        int32_t calendarCount = MIN(calendarIdentifiers.count, calendarsCapacity);
+        int32_t calendarIndex = 0;
+        calendars[calendarIndex++] = GetCalendarId([defaultCalendarIdentifier UTF8String]);
```

Comment:
Theres a possibility of this being `UNINITIALIZED_VALUE` right? It doesn't seem like we should be setting this if its not a known calendar Id. Which highlights that there are more calendars in the `calendarIdentifiers` list than in the known calendarIds in `GetCalendarId`. Should we expand the coverage of `GetCalendarId` and `GetCalendarIdentifier`?

Reply:
Nice catch, I will update it.

Reply:
Updated the part related to `UNINITIALIZED_VALUE` , but for expanding coverage in `GetCalendarId` and `GetCalendarIdentifier`,  looks like it is not smth that we support in C# side with current implementation. Supported calendars are implemented in C# like for example https://github.com/mkhamoyan/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.cs . I guess I can open a ticket to see if we can implement more calendars that are supported by `NSCalendarIdentifier`. What do you think?